packages/plugins/lib/interfaces.ts

```diff
@@ -0,0 +1,31 @@
+// plugin configuration as specified in projects' truffle-config.js
+// supports either plain module name as string, or wrapper object
+export type RawPluginConfig = PluginConfig | string;
+
+// internal representation of a particular plugin configuration
+export interface PluginConfig {
+  module: string;
+}
+
+// Partial representation of the truffle-config object
+// TODO: Properly define the full interface for this object in the appropriate module and import it here
+export interface TruffleConfig {
```

Comment:
So @truffle/config does export types - does it make sense to use that type directly?

Reply:
It looks like the `TruffleConfig` exported in `@truffle/config` doesn't have very strong typings:

```ts
declare class TruffleConfig {
    [key: string]: any;
    private _deepCopy;
    private _values;
    constructor(truffleDirectory?: string, workingDirectory?: string, network?: any);
    eventManagerOptions(config: TruffleConfig): any;
    addProp(propertyName: string, descriptor: any): void;
    normalize(obj: any): any;
    with(obj: any): TruffleConfig;
    merge(obj: any): TruffleConfig;
    static default(): TruffleConfig;
    static search(options?: any, filename?: string): string | null;
    static detect(options?: any, filename?: string): TruffleConfig;
    static load(file: string, options?: any): TruffleConfig;
    static getUserConfig(): Configstore;
    static getTruffleDataDirectory(): string;
}
```

So it doesn't tell you that e.g. `working_directory` is an existing key with a `string` value type. It might just be fine to go with this less strict type, but it might also be good to add stricter typings to the `@truffle/config` package at some point.

Reply:
Sounds good - let's keep it lenient for now!